### PR TITLE
Creation phase

### PR DESCRIPTION
**Pull request by VeeAndrade**
#### What's this PR do?
This PR creates all the test files and the class files for this project. I have added basic testing for class files and started on my javascript functionality. Along with this, I attempted to have a multipage app by configuring webpack that allow me to have multiple entry points with javascript. I have also built out the html skeleton and the styles for most of the pages. 
#### Where should the reviewer start?
Anywhere really, all the files are new 

@khalidwilliams

### CODE REVIEW

File: src/css/base.scss [hunk 1]
```diff
@@ -1,6 +1,128 @@
...
+  width: 100%;
+  height: 20%;
+  border-radius: 7px;
+  padding-left: 10px;
+  outline: none;
+  border: none;
+}
+
+.user-password-bar {
+  @extend .user-input-bar;
```
Comment (joel-oe-lacey) [inline src/css/base.scss]: I don't know if you plan to add additional properties to .user-password-bar, if not you could just add that class onto the .user-input-bar selector and comma separate, the extend might be extra.

File: src/css/user.scss [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+  @extend .header-navbar;
+}
+
+.client-header-heading {
+  @extend .header-heading;
+}
+
+.client-header-navbar-button {
+  @extend .header-navbar-button
+}
```
Comment (joel-oe-lacey) [inline src/css/user.scss]: Same as above on these three.

File: src/main-client.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
+import $ from 'jquery'
+import moment from 'moment'
+import Hotel from './Hotel';
+import Booking from './Booking';
+import Room from './Room';
+
+let url, urlString, userId, hotel
```
Comment (joel-oe-lacey) [inline src/main-client.js]: Couple places where semicolons are missing, not breaking, but probably stylistically worth adding.

File: src/main-client.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      hotel = new Hotel(rooms, bookings);
+      welcomeUser(user.name)
+      let userBookings = hotel.getUserBookings(user.id)
+      let todaysDate = moment().format('YYYY/MM/DD')
+      $('.booking-input-field').attr('min', todaysDate.split('/').join('-'))
+      getPastBookingsOnDom(userBookings, hotel.rooms, todaysDate)
+    })
+}
+
+if (window.location.href.includes("userId")) {
```
Comment (joel-oe-lacey) [inline src/main-client.js]: I like the URL transfer approach, cool clever implementation!

File: src/main-client.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  getUserDashBoard(userId)
+  // $('.booking-date-button').on('click', filterRooms)
+}
+
+function welcomeUser(name) {
+  $('.client-welcome-heading').text(`Welcome ${name.split(' ')[0]}`)
+}
+
+function getPastBookingsOnDom(bookings, rooms, todaysDate) {
+  let upcomingBookings = bookings.filter(booking => booking.date >= todaysDate)
```
Comment (joel-oe-lacey) [inline src/main-client.js]: As it pertains to bookings, might fit better as a method on the class.

Comment (joel-oe-lacey) [reply src/main-client.js]: Same with calculateBookingCosts below

Comment (VeeAndrade) [reply src/main-client.js]: I will definitely try to refactor this and add these functions to the appropriate classes.

File: src/css/base.scss (hunk 1 above)
Comment (VeeAndrade) [reply src/css/base.scss]: I completely understand and will probably be refactoring this as the last step

**Pull request merged**